### PR TITLE
[Move] Fully Implement Plasma Fists

### PR DESCRIPTION
**Pull request by innerthunder**
## What are the changes the user will see?
This implements the secondary effect of [Plasma Fists](https://bulbapedia.bulbagarden.net/wiki/Plasma_Fists_(move)), which converts all Normal-type moves to Electric type for the rest of the turn.

## Why am I making these changes?
Less (P)

## What are the changes from a developer perspective?
- `data/arena-tag`: New `PlasmaFistsTag` converts Normal type to Electric type. It is always applied to both sides of the field.
- `data/move`: Added attribute to Plasma Fists that adds a `PlasmaFistsTag` to the field. Plasma Fists is no longer marked with `.partial()`
- `field/pokemon`: `Pokemon.getMoveType` now applies Plasma Fists' field effect after all other type change effects are applied.
- `test/utils/gameManager`: Fixed a bug with `forceEnemyMove` where the function couldn't strictly force the enemy to target the first player Pokemon (`BattlerIndex.PLAYER`)
- `test/moves/plasma_fists.test`: New unit tests on Plasma Fist's functionality. Includes tests for:
  - Normal-type move conversion on both player and enemy attacks in a double battle
  - Interaction with Pixilate (Normal moves become Fairy, not Electric)
  - Interaction with Normalize (Normalized non-Normal moves become Electric)

## How to test the changes?
`npm run test plasma_fists`

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by DayKev — APPROVED** [review]

**Review by Adri1 — APPROVED** [review]

**Comment by Adri1** [comment]
All languages implemented

File: src/test/moves/plasma_fists.test.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    game.move.select(Moves.PLASMA_FISTS);
+
+    await game.setTurnOrder([BattlerIndex.PLAYER, BattlerIndex.ENEMY]);
+    await game.phaseInterceptor.to("BerryPhase", false);
+
+    expect(enemyPokemon.getMoveType).toHaveLastReturnedWith(Type.FAIRY);
+    expect(playerPokemon.hp).toBeLessThan(playerPokemon.getMaxHp());
+  });
+
+  it("should affect moves that become Normal type due to Normalize", async () => {
```
Comment (Tempo-anon) [inline src/test/moves/plasma_fists.test.ts:78]: What about the edge case of a Pokemon with both normalize and pixilate? 🤔

Comment (DayKev) [reply src/test/moves/plasma_fists.test.ts:78]: Seems like those moves are Fairy type (just tested). That situation isn't really possible to happen in the official games so... that's probably fine?

**Review by DayKev — APPROVED** [review]

File: src/test/moves/plasma_fists.test.ts (hunk 1 above)
Comment (innerthunder) [reply src/test/moves/plasma_fists.test.ts:78]: In theory it depends on which is the base ability and which is the passive, since `applyAbAttrsInternal` applies base abilities first. So if a Pokemon's base/passive is Normalize/Pixilate, all attacks would be Fairy type, but when switched around to Pixilate/Normalize, all attacks would be Normal type (which could then change to Electric with Plasma Fists' effect active).

In any case, Plasma Fists' effect applies last in `getMoveType`, which is consistent with how it works in mainline.

File: src/data/arena-tag.ts [hunk 2]
```diff
@@ -511,6 +511,39 @@ class WaterSportTag extends WeakenMoveTypeTag {
   }
 }
 
+/**
+ * Arena Tag class for the secondary effect of {@link https://bulbapedia.bulbagarden.net/wiki/Plasma_Fists_(move) | Plasma Fists}.
+ * Converts Normal-type moves to Electric type for the rest of the turn.
+ */
+export class PlasmaFistsTag extends ArenaTag {
```
[the remark this answers is not in the document]
Comment (innerthunder) [reply src/data/arena-tag.ts:518]: I looked into it, but there are a couple differences between the moves that might make reuse difficult:
- Electrify only affects a single target, so it's probably better implemented as a battler tag.
- Electrify applies to any move, not just Normal-type moves

[the remark this answers is not in the document]
Comment (Tempo-anon) [reply src/data/arena-tag.ts:518]: My initial thoughts were to have a single battlertag something like
```ts
// Used for electrify (electrifyAllTypes = true) and plasma fists (electrifyAllTypes = false)
export class ElectrifiedTag extends BattlerTag {
  electrifyAllTypes: boolean;
}
```
where Electrify would have all types and plasma fist would just have normal and plasma fist would just apply the battler tag to all pokemon on the field.

[the remark this answers is not in the document]
Comment (innerthunder) [reply src/data/arena-tag.ts:518]: What would happen if both moves were used in the same turn? My only hangup really is having to deal with overlap logic to set `electrifyAllTypes` correctly (on-add messages are a little annoying too, but manageable)

[the remark this answers is not in the document]
Comment (Tempo-anon) [reply src/data/arena-tag.ts:518]: Could do some boolean logic to set the `electrifyAllTypes` to be `OR` result of the tags. That and the message can be done by just modifying the `onAdd` function like how Autotomize only displays the message if weight was reduced.
